File: 2500-delete-greatest-value-in-each-row/2500-delete-greatest-value-in-each-row.java
class Solution {
    public int deleteGreatestValue(int[][] grid) {
        int r=grid.length;
        int c=grid[0].length;
        int res=0;
        //O(nlogn) O1
        
        for(int i=0;i<r;i++)
            Arrays.sort(grid[i]);   //rows got sorted 
        
        for(int j=0;j<c;j++){   
            int mx=Integer.MIN_VALUE;
            for(int i=0;i<r;i++){   //traverse each rows
                mx=Math.max(mx,grid[i][j]); //will get max  value from each rows
            }
            res+=mx;    
        }
        return res;
    }
}

File: 2500-delete-greatest-value-in-each-row/2500-delete-greatest-value-in-each-row.cpp
class Solution {
public:
    int deleteGreatestValue(vector<vector<int>>& grid) 
    {
        int n=grid[0].size();
        int ans=0;
        int r=grid.size();
        vector<priority_queue<int>> pq(r);
        int c=grid[0].size();
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                pq[i].push(grid[i][j]);
            }
        }
    
        while(n--)
        {
            int tmp=INT_MIN;
            for(int i=0;i<r;i++)
            {
                tmp=max(tmp,pq[i].top());
                pq[i].pop();
            }
            ans+=tmp;
        }
        return ans;
        
    }
};